projects/packages/forms/src/contact-form/class-contact-form.php

```diff
@@ -1741,4 +1741,95 @@ public static function get_block_alignment_class( $attributes = array() ) {
 		}
 		return $align_to_class_map[ $attributes['align'] ];
 	}
+
+	/**
+	 * Get the form's accessible name
+	 *
+	 * @param string $content The form's content.
+	 * @return string The form's accessible name.
+	 */
+	public function get_form_label( $content ) {
+		global $post;
+
+		$headings = array_map(
+			function ( $i ) {
+				return "h$i";
+			},
+			range( 1, 6 )
+		);
+
+		// 1. Look for a heading inside the form
+
+		$dom = new \DOMDocument();
+		$dom->loadHTML( $content );
+
+		foreach ( $headings as $heading ) {
```

Comment:
Maybe refactor this into a method since we are using it in Step 2 as well.
```php
public function get_aria_label_based_on_headings(DOMDocument $dom, array $headings): string {
        foreach ($headings as $heading) {
            $heading_element = $dom->getElementsByTagName($heading)->item(0);

            if (isset($heading_element)) {
                $id = $heading_element->getAttribute('id');

                if (isset($id) && !empty($id)) {
                    return "aria-labelledby='$id'";
                }

                return "aria-label='" . $heading_element->textContent . "'";
            }
        }
        return '';
    }
```
And then check for it not being empty to return.